### PR TITLE
[new downloader] Fix calling callbacks for country being deleted

### PR DESCRIPTION
**Pull request by syershov**
MAPSME-202

@ygorshenin @gardster PTAL
Восстанавливает утерянное обновление модели при удалении мвм.

### CODE REVIEW

File: storage/storage.cpp [hunk 1]
```diff
@@ -127,11 +127,12 @@ Storage::Storage(string const & referenceCountriesTxtJsonForTesting,
   CHECK_LESS_OR_EQUAL(0, m_currentVersion, ("Can't load test countries file"));
 }
 
-void Storage::Init(TUpdate const & update)
+void Storage::Init(TUpdate const & didDownload, TUpdate const & willDelete)
```
Comment (ygorshenin) [inline storage/storage.cpp]: s/didDownload/onDownloaded/g
s/willDelete/onDeleted/g

File: storage/storage.hpp [hunk 2]
```diff
@@ -106,7 +106,8 @@ class Storage
 {
 public:
   struct StatusCallback;
-  using TUpdate = function<void(platform::LocalCountryFile const &)>;
+  using TLocalFilePtr = shared_ptr<platform::LocalCountryFile>;
+  using TUpdate = function<void(storage::TCountryId const &, TLocalFilePtr const)>;
```
Comment (ygorshenin) [inline storage/storage.hpp]: s/TCountryChanged/g

Comment (ygorshenin) [inline storage/storage.hpp]: TOnCountryChanged

File: map/framework.cpp [hunk 3]
```diff
@@ -464,20 +466,38 @@ void Framework::ShowNode(storage::TCountryId const & countryId)
...
+    auto p = m_model.RegisterMap(*localFile);
+    MwmSet::MwmId const & id = p.first;
+    if (id.IsAlive())
+      rect = id.GetInfo()->m_limitRect;
+  }
+  InvalidateRect(rect);
+  m_searchEngine->ClearCaches();
+}
 
+void Framework::OnCountryFileDelete(storage::TCountryId const & countryId, storage::Storage::TLocalFilePtr const localFile)
```
Comment (ygorshenin) [inline map/framework.cpp]: s/OnCountryFileDelete/OnCountryFileDeleted/g

**Comment by ygorshenin** [comment]
Seems good, but please, fix naming.

**Comment by gardster** [comment]
LGTM

**Comment by syershov** [comment]
@ygorshenin PTAL

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**